docs/hyperopt.md

```diff
@@ -1,153 +1,109 @@
-        if params['rsi']['enabled']:
-            conditions.append(dataframe['rsi'] < params['rsi']['value'])
-
-        # TRIGGERS
-        triggers = {
-            'lower_bb': dataframe['tema'] <= dataframe['blower'],
-            'faststoch10': (crossed_above(dataframe['fastd'], 10.0)),
-            'ao_cross_zero': (crossed_above(dataframe['ao'], 0.0)),
-            'ema5_cross_ema10': (crossed_above(dataframe['ema5'], dataframe['ema10'])),
-            'macd_cross_signal': (crossed_above(dataframe['macd'], dataframe['macdsignal'])),
-            'sar_reversal': (crossed_above(dataframe['close'], dataframe['sar'])),
-            'stochf_cross': (crossed_above(dataframe['fastk'], dataframe['fastd'])),
-            'ht_sine': (crossed_above(dataframe['htleadsine'], dataframe['htsine'])),
-        }
-        ...
 ```
-
-
-### 2. Update the hyperopt config file
-Hyperopt is using a dedicated config file. Currently hyperopt 
-cannot use your config file. It is also made on purpose to allow you
-testing your strategy with different configurations.
-
-The Hyperopt configuration is located in 
-[user_data/hyperopt_conf.py](https://github.com/freqtrade/freqtrade/blob/develop/user_data/hyperopt_conf.py).
-
-
-## Advanced notions
-### Understand the Guards and Triggers
-When you need to add the new guards and triggers to be hyperopt 
-parameters, you do this by adding them into the [hyperopt_space()](https://github.com/freqtrade/freqtrade/blob/develop/user_data/strategies/test_strategy.py#L244-L297).
-
-If it's a trigger, you add one line to the 'trigger' choice group and that's it.
-
-If it's a guard, you will add a line like this:
-```
-'rsi': hp.choice('rsi', [
-    {'enabled': False},
-    {'enabled': True, 'value': hp.quniform('rsi-value', 20, 40, 1)}
-]),
+    def indicator_space() -> List[Dimension]:
+        """
+        Define your Hyperopt space for searching strategy parameters
+        """
+        return [
+            Integer(20, 40, name='adx-value'),
+            Integer(20, 40, name='rsi-value'),
+            Categorical([True, False], name='adx-enabled'),
+            Categorical([True, False], name='rsi-enabled'),
+            Categorical(['bb_lower', 'macd_cross_signal'], name='trigger')
+        ]
 ```
-This says, "*one of the guards is RSI, it can have two values, enabled or 
-disabled. If it is enabled, try different values for it between 20 and 40*".
 
-So, the part of the strategy builder using the above setting looks like 
-this:
+Above definition says: I have five parameters I want you to randomly combine 
+to find the best combination. Two of them are integer values (`adx-value` 
+and `rsi-value`) and I want you test in the range of values 20 to 40.
+Then we have three category variables. First two are either `True` or `False`.
+We use these to either enable or disable the ADX and RSI guards. The last
+one we call `trigger` and use it to decide which buy trigger we want to use.
```

Comment:
more a question for my understanding - can the result also a combination of "both" - or an that be configured (say i want "bb_lower and macd_crossed", or "bb_lower, crossed_ema" ?

Reply:
For triggers, it currently can't be a combination. Currently it just picks one from the trigger list. We could definitely add a more complex "pick at least one trigger" there.